enhancements/migration/cam-non-admin-design.md

```diff
@@ -0,0 +1,351 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+The risks of implementing this is if a user Bob is capable of migrating
+resources that he does not have proper access to on the source or destination
+clusters. Since a migration includes a backup and a restore on the respective
+clusters, Bob is performing privileged operations that should require some form
+of authorization of the targeted resources before performing the operation.
+
+In order to mitigate this, we are ensuring the design is vetted by the
+appropiate teams that can ensure we are doing what is normal practice in other
+OCP items like the `projects` API.
+
+## Design Details
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+
+##### Setup
+
+Source cluster is OCP3, target cluster is OCP4 and the target cluster is also
+the host cluster.
+
+##### Workflow
+
+1. Mig operator is installed via cluster-admin by OLM in `openshift-migration`
+   namespace on target/source/host clusters. `openshift-migration` namespace
+   should be only accessible by the cluster-admin.
+  1. Operator sets up `mig` service accounts on each cluster in
+     `openshift-migration` with a rolebinding which grants that service account
+     all the permissions required to perform a migration.
+1. The cluster-administrator creates secrets in `openshift-migration` which
+   contains the SA token of the mig service account for both the source and
+   target cluster. (In this example, since the target is the host cluster this
+   will be configured for the cluster-admin, so he only needs to create the one
+   secret).
+1. The cluster-administrator then grants Bob “USE” on these created secret(s)
```

Comment:
Can we go into more detail why it is not sufficient to use this as the ok that the user should have access to escalation? 

IIRC the reason is Valero objects, which is what this would grant the user access to escalate to use, would allow you to migrate the whole cluster even if you only had access,s to a single namespace. While we can use a different SA based on user for auditing purposes, this does not give users the safety that an admin would expect, and that is why CAM must have a mechanism to test the user's permissions in the source cluster. Is that about right?

Reply:
That is correct. Granting the user this escalation grants them the ability to create backup/restore Velero resources. If we only depend on this to check if a user can perform a migration, then the user could specify namespaces to backup which they do not have access to. I will add a small blurb about this